### DISCUSSION TITLE
API Outage?

### DISCUSSION THREAD

**Discussion by rustyshelf** [amberelectric/public-api#130] [General]
My app has been like this for a few hours now, and the only other person I know on Amber in Adelaide has the same issue. Is there an outage?

![IMG_0468](https://user-images.githubusercontent.com/610217/219330455-cbde5f68-1f77-48a4-bca5-a8e4ce01baaa.png)

**Comment by madpilot** [comment]
Yes, we got hit by a rate limit that caused a failure cascade. We have just reverted the change that caused the problem - we'll be back at 10pm (which is when the limit gets reset).

**Reply by rustyshelf** [reply]
Fun time to do it. Instead of making $40-$80 like the previous few spikes, I have a cost of $1 );

**Reply by madpilot** [reply]
We'll be reviewing tomorrow, to work out how to not have this happen again.

**Reply by icenov** [reply]
Not sure if this is related, but we've not had a usage update since Tuesday? NSW far south coast area.

**Reply by chiefcomm** [reply]
> Not sure if this is related, but we've not had a usage update since Tuesday? NSW far south coast area.

@icenov good to know it is more than just me having no update since Tuesday :-)

**Reply by madpilot** [reply]
It's not. It's related to a different issue with our usage provider. It's been escalated, we are waiting for a response from them

**Reply by rustyshelf** [reply]
> It's related to a different issue with our usage provider.

I've always been curious, is that from SA Power Networks, or do you have a third party that does it for all of Australia that you use?

> We'll be reviewing tomorrow, to work out how to not have this happen again.

No worries, I've been in this industry long enough to know that it happens to the best of us, so I'm torn between being supportive vs the in built expectation that a utility like your power retailer should never go down ;). Either way, I'm sure it will lead to more thorough testing before a deployment which is always a good thing.

**Reply by madpilot** [reply]
Message we've sent to everyone this morning

> Dear Amber users
>
> Reaching out here as we had an issue with our automation and communications systems starting at around 6.30-7pm last night.
>
> We designated it a critical engineering incident and our team logged back in and were able to fix the bug by 11:30pm.
>
> Unfortunately, this occurred during an evening of increased volatility, when it's critical that our systems are working and helping you to be able to maximise the value of your battery. This is the first time we've seen an issue like this during a critical period.
>
> We're building at a high pace to continue to expand the functionality of our product for you, but that shouldn't come at the expense of reliability when you need it. We're extremely sorry for letting you down and we'll be doing a thorough investigation to understand the root cause and avoid it happening again.
>
> For impacted customers, we will be waiving our subscription fee for February and March as we haven't delivered what we promised during this period. Please reach out if you have any questions.
>
> Regards,
> Chris and the Amber team
>
> (Note - we've also discovered an unrelated issue with meter data availability for Weds and Thurs - this is caused by an issue with our upstream data provider. We're working with them to get this resolved and expect to be fixed in the next few hours)

**Reply by madpilot** [reply]
> I've always been curious, is that from SA Power Networks, or do you have a third party that does it for all of Australia that you use?

Our billing provider does it on our behalf. 

Traditionally, reading meters was the responsibility of the network, but will the Smartmeter roll out, it is done by a metering provider (except in Victoria - it is still done by the network). There is a common system called MSATS that smart meter providers dump data into, and everyone else reads out of. We only just got access to MSATS after we go our retail licence, so we haven't spun up the project to read directly yet, so we just use the feed our billing provider supplies us.

**Reply by rob0101** [reply]
Please tell me you don’t still use Energy Locals for billing. 

I can find a significant error in every bill they have input into. 

Been with them via Amber (previously), EL directly and IO Energy.  They ****ed up every bill they touched.

**Reply by madpilot** [reply]
We don't use Energy Locals for billing.
